compiler/src/iree/compiler/Dialect/VM/Conversion/VMToEmitC/test/control_flow_ops.mlir

```diff
@@ -668,58 +678,56 @@ vm.module @my_module {
-  // CHECK-NEXT: emitc.assign %[[RESULTSIZE]] : !emitc.opaque<"iree_host_size_t"> to %[[RESSDATALENGTH]] : !emitc.opaque<"iree_host_size_t">
-  // CHECK-NEXT: %[[RESSDATA:.+]] = "emitc.member_of_ptr"(%[[RESBYTESPAN]]) <{member = "data"}> : (!emitc.ptr<!emitc.opaque<"iree_byte_span_t">>) -> !emitc.ptr<ui8>
-  // CHECK-NEXT: emitc.assign %[[RESBYTESPANDATA]] : !emitc.ptr<ui8> to %[[RESSDATA]] : !emitc.ptr<ui8>
+  // CHECK-NEXT: %[[RESSDATALENGTH:.+]] = "emitc.member"(%[[RESBYTESPAN_MEMBER]]) <{member = "data_length"}> : (!emitc.lvalue<!emitc.opaque<"iree_byte_span_t">>) -> !emitc.lvalue<!emitc.opaque<"iree_host_size_t">>
+  // CHECK-NEXT: emitc.assign %[[RESULTSIZE]] : !emitc.opaque<"iree_host_size_t"> to %[[RESSDATALENGTH]] : <!emitc.opaque<"iree_host_size_t">>
+  // CHECK-NEXT: %[[RESSDATA:.+]] = "emitc.member"(%[[RESBYTESPAN_MEMBER]]) <{member = "data"}> : (!emitc.lvalue<!emitc.opaque<"iree_byte_span_t">>) -> !emitc.lvalue<!emitc.ptr<ui8>>
+  // CHECK-NEXT: emitc.assign %[[RESBYTESPANDATA]] : !emitc.ptr<ui8> to %[[RESSDATA]] : <!emitc.ptr<ui8>>
   // CHECK-NEXT: emitc.call_opaque "memset"(%[[RESBYTESPANDATA]], %[[RESULTSIZE]]) {args = [0 : index, 0 : ui32, 1 : index]}
 
   // Pack the argument into the struct.
-  // CHECK-NEXT: %[[ARGS:.+]] = "emitc.variable"() <{value = #emitc.opaque<"">}> : () -> !emitc.opaque<"iree_byte_span_t">
-  // CHECK-NEXT: %[[ARGS_MEMBER:.+]] = "emitc.member"(%[[ARGSTRUCT]]) <{member = "arguments"}> : (!emitc.opaque<"iree_vm_function_call_t">) -> !emitc.opaque<"iree_byte_span_t">
-  // CHECK-NEXT: emitc.assign %[[ARGS_MEMBER]] : !emitc.opaque<"iree_byte_span_t"> to %[[ARGS]] : !emitc.opaque<"iree_byte_span_t">
-  // CHECK-NEXT: %[[ARGSPTR:.+]] = "emitc.variable"() <{value = #emitc.opaque<"">}> : () -> !emitc.ptr<ui8>
-  // CHECK-NEXT: %[[ARGSPTR_MEMBER:.+]] = "emitc.member"(%[[ARGS]]) <{member = "data"}> : (!emitc.opaque<"iree_byte_span_t">) -> !emitc.ptr<ui8>
-  // CHECK-NEXT: emitc.assign %[[ARGSPTR_MEMBER]] : !emitc.ptr<ui8> to %[[ARGSPTR]] : !emitc.ptr<ui8>
-  // CHECK-NEXT: %[[ARG:.+]] = emitc.cast %[[ARGSPTR]] : !emitc.ptr<ui8> to !emitc.ptr<!emitc.opaque<"iree_vm_ref_t">>
+  // CHECK-NEXT: %[[ARGS:.+]] = "emitc.member"(%[[ARGSTRUCT]]) <{member = "arguments"}> : (!emitc.lvalue<!emitc.opaque<"iree_vm_function_call_t">>) -> !emitc.lvalue<!emitc.opaque<"iree_byte_span_t">>
+  // CHECK-NEXT: %[[ARGSDATA_LVAL:.+]] = "emitc.member"(%[[ARGS]]) <{member = "data"}> : (!emitc.lvalue<!emitc.opaque<"iree_byte_span_t">>) -> !emitc.lvalue<!emitc.ptr<ui8>>
+  // CHECK-NEXT: %[[ARGSDATA:.+]] = emitc.load %[[ARGSDATA_LVAL]] : <!emitc.ptr<ui8>>
+  // CHECK-NEXT: %[[REF_LVAL:.+]] = "emitc.variable"() <{value = #emitc.opaque<"">}> : () -> !emitc.lvalue<!emitc.ptr<!emitc.opaque<"iree_vm_ref_t">>>
+  // CHECK-NEXT: emitc.assign %arg2 : !emitc.ptr<!emitc.opaque<"iree_vm_ref_t">> to %[[REF_LVAL]] : <!emitc.ptr<!emitc.opaque<"iree_vm_ref_t">>>
```

Comment:
```suggestion
```
With https://github.com/iree-org/iree/pull/18286/commits/6cd72cb81cbb3df10db379c14cb4a08a038d06f5, this is no longer emitted. @simon-camp please take a look as well.

Reply:
There are some more fixes needed. Are you okay if I directly push the commits to your branch @Groverkss?